ExtraDry/Sample.Data/Extensions/TaxonomyExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using System;
+
+namespace Sample.Data {
+    public static class TaxonomyExtensions {
+        /// <summary>
+        /// Moves a subtree from one point in the heirarchy to another.
+        /// </summary>
+        public static async Task MoveSubtree(this ITaxonomyEntity subtreeRootToMove, ITaxonomyEntity newParent, DbContext context)
+        {
+            // Check that the new parent is at the same strata level as the current parent.
+            if(subtreeRootToMove.Strata != newParent.Strata + 1) {
+                throw new ArgumentException("Moving the subtree cannot change it's strata level");
+            }
+
+            // This should align with the name of the closure table due to EF convention. eg. Region -> RegionRegion, Location -> LocationLocation
+            var closureTableName = $"[{subtreeRootToMove.GetType().Name}{subtreeRootToMove.GetType().Name}]";
+
+            // Query based off the closure table move-subtree queries as found in the book SQL Antipatterns by Bill Karwin
+            // First step deletes references from the super tree to items within the subtree, retaining the self-reference of the root node.
+            await context.Database.ExecuteSqlRawAsync($@"DELETE FROM {closureTableName}
```

Comment:
This feels like it should be wrapped in a transaction.

Reply:
It is now.